tools/testdebug/cmd/load_etcd.go

```diff
@@ -0,0 +1,172 @@
+	AllowAll     bool
+}
+
+func NewDebugAPIServerCommand() *cobra.Command {
+	o := &DebugAPIServerOptions{Out: os.Stdout}
+
+	cmd := &cobra.Command{
+		Use:   RecommendedLoadEtcdDumpName + " etcd_dump.json",
+		Short: "Start API server using etcddump",
+		Run: func(cmd *cobra.Command, args []string) {
+			kcmdutil.CheckErr(o.Complete(args))
+
+			kcmdutil.CheckErr(o.Run())
+		},
+	}
+
+	cmd.Flags().BoolVar(&o.AllowAll, "allow-all", true, "change policy to grant system:authenticated cluster-admin powers")
+
+	flagtypes.GLog(cmd.PersistentFlags())
+
+	return cmd
+}
+
+func (o *DebugAPIServerOptions) Complete(args []string) error {
+	if len(args) != 1 {
+		return errors.New("etcd_dump.json file is required")
+	}
+
+	o.EtcdDumpFile = args[0]
+
+	return nil
+}
+
+func (o *DebugAPIServerOptions) Run() error {
+	masterConfig, err := testserver.DefaultMasterOptionsWithTweaks(true, true)
+	if err != nil {
+		return err
+	}
+
+	etcdConfig := masterConfig.EtcdConfig
+	masterConfig.EtcdConfig = nil
+	masterConfig.DNSConfig = nil
+
+	etcdserver.RunEtcd(etcdConfig)
+
+	o.ImportEtcdDump(masterConfig.EtcdClientInfo)
+
+	o.StartAPIServer(*masterConfig)
+
+	if o.AllowAll {
+		osClient, err := testutil.GetClusterAdminClient(testutil.GetBaseDir() + "/openshift.local.config/master/admin.kubeconfig")
+		if err != nil {
+			return err
+		}
+
+		addClusterAdmin := &policy.RoleModificationOptions{
+			RoleName:            bootstrappolicy.ClusterAdminRoleName,
+			RoleBindingAccessor: policy.ClusterRoleBindingAccessor{Client: osClient},
+			Groups:              []string{"system:authenticated"},
+		}
+		if err := addClusterAdmin.AddRole(); err != nil {
+			return err
+		}
+	}
+
+	// trick govet
+	if 0 == 0 {
+		select {}
+	}
+
+	return nil
+}
+
+func (o *DebugAPIServerOptions) StartAPIServer(masterConfig configapi.MasterConfig) error {
+	openshiftConfig, err := origin.BuildMasterConfig(masterConfig)
+	if err != nil {
+		return err
+	}
+
+	kubeMasterConfig, err := start.BuildKubernetesMasterConfig(openshiftConfig)
+	if err != nil {
+		return err
+	}
+
+	fmt.Printf("Starting master on %s\n", masterConfig.ServingInfo.BindAddress)
+	fmt.Printf("Public master address is %s\n", masterConfig.AssetConfig.MasterPublicURL)
+	if err := start.StartAPI(openshiftConfig, kubeMasterConfig); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (o *DebugAPIServerOptions) ImportEtcdDump(etcdClientInfo configapi.EtcdConnectionInfo) error {
+	etcdDumpContent, err := ioutil.ReadFile(o.EtcdDumpFile)
+	if err != nil {
+		return err
+	}
+	etcdDump := &coreosetcdclient.Response{}
+	if err := json.Unmarshal(etcdDumpContent, etcdDump); err != nil {
+		return err
+	}
+
+	// Connect and setup etcd interfaces
+	etcdClient, err := etcd.GetAndTestEtcdClient(etcdClientInfo)
+	if err != nil {
+		return err
+	}
+
+	i := -1
+	nodeList := []*coreosetcdclient.Node{}
+	nodeList = append(nodeList, etcdDump.Node)
+	for {
+		i++
+		if i >= len(nodeList) {
+			break
+		}
+		node := nodeList[i]
```

Comment:
I understand why you're not doing `:= range nodeList`... but why not `for i :=0; i<len(nodeList); i++` ?